# FAQ

## BEAST2

### Which version of BEAUti do you use as a guideline?

Run this code:

```
beastier::get_beast2_version()
```

Currently, this returns `2.6.4` which means that the
BEAUti version `2.6` (the patch version is irrelevant)
is ignored.

### How to install BEAST2?

See [install.md](install.md)

## `babette` development 

### What's the road map?

Currently, `babette` does not have a road map itself, but `beautier` does:

 * [beautier road map](https://github.com/ropensci/beautier/blob/main/doc/road_map.md)

### How can I indicate a feature that I miss?

Submit an Issue.

### How can I submit code?

See [CONTRIBUTING](../CONTRIBUTING.md), at 'Submitting code'

### How can I submit a bug?

See [CONTRIBUTING](../CONTRIBUTING.md), at 'Submitting bugs' 

### How can I indicate something else?

Submit an Issue. Or send an email to Richèl Bilderbeek.

### What are the `babette` dependencies?

![babette dependencies](dependencies.png)

## `babette` technical questions

### How can I inspect a generated BEAST2 `.xml` file?

The path to the BEAST2 `.xml` file, 
which is the file that contains the inference model,
can be found in the BEAST2 options.
A typical user is unaware these options exists,
as these are generated by default.

Here is an example that calls `bbt_run_from_model` with
an explicit BEAST2 options:

```
# Default BEAST2 options
beast2_options <- create_beast2_options()

# The path where the BEAST2 XML file will be created:
beast2_options$input_filename

bbt_run_from_model(
  fasta_filename = get_babette_path("anthus_aco.fas"),
  beast2_options = beast2_options
)

# Show BEAST2 XML file
readLines(beast2_options$input_filename)
```

### How can I inspect a generated BEAST2 `.xml.state` file?

The path to the BEAST2 `.xml.state` file, 
which is the file that contains the current/final state of the MCMC machinery,
can be found in the BEAST2 options.
A typical user is unaware these options exists,
as these are generated by default.

A typical user is unaware these options exists,
as these are generated by default.

Here is an example that calls `bbt_run_from_model` with
an explicit BEAST2 options:

```
# Default BEAST2 options
beast2_options <- create_beast2_options()

# The path where the final BEAST2 state file will be created:
beast2_options$output_state_filename

bbt_run_from_model(
  fasta_filename = get_babette_path("anthus_aco.fas"),
  beast2_options = beast2_options
)

# Show BEAST2 state file
readLines(beast2_options$output_state_filename)
```

### How can I inspect a generated BEAST2 `.trees` file?

This `.trees` file, which holds the estimated trees,
is part of the `inference_model`:

```
inference_model <- create_test_inference_model()

# Path to .trees file
inference_model$mcmc$treelog$filename

bbt_run_from_model(
  fasta_filename = get_babette_path("anthus_aco.fas"),
  inference_model = inference_model
)

# Show the generated .trees file
readLines(inference_model$mcmc$tracelog$filename)
```

### How can I inspect a generated BEAST2 `.log` file?

This `.log` file holds the parameter estimations
is part of the `inference_model`:

```
inference_model <- create_test_inference_model()

# Path to .log file
inference_model$mcmc$tracelog$filename

bbt_run_from_model(
  fasta_filename = get_babette_path("anthus_aco.fas"),
  inference_model = inference_model
)

# Show the generated .log file
readLines(inference_model$mcmc$tracelog$filename)
```

### Can `babette` handle missing data in the FASTA files?

Yes. BEAST2 can missing data in the FASTA files (which
contain the DNA/RNA or protein sequences) and so does `babette`.

In your FASTA file, use a dash `-` (not NA)
for the sequences that are missing.
This is similar to what the `phangorn` package does.

### If I set a fixed crown age with multiple alignments, only the first alignment has so

Correct. This is a feature of BEAST2. Using the `create_mrca` prior 
gives prettier results.

## `babette` in academia

### How do I reference to this work?

```
Bilderbeek, Richèl JC, and Rampal S. Etienne. "babette: BEAUti 2, BEAST 2 and Tracer for R." Methods in Ecology and Evolution (2018).
```

or

```
@article{bilderbeek2018babette,
  title={babette: BEAUti 2, BEAST 2 and Tracer for R},
  author={Bilderbeek, Richel JC and Etienne, Rampal S},
  journal={Methods in Ecology and Evolution},
  year={2018},
  publisher={Wiley Online Library}
}
```

### Are there any related packages?

 * [lumier](https://github.com/richelbilderbeek/lumier): Shiny app to help create the function call needed
 * [BEASTmasteR](https://github.com/nmatzke/BEASTmasteR): tip-dating analyses using fossils as dated terminal taxa
 * [RBeast](https://github.com/beast-dev/RBeast): misc other things


### What are the FASTA files?

FASTA files `anthus_aco.fas` and `anthus_nd2.fas` from:
 
 * Van Els, Paul, and Heraldo V. Norambuena. "A revision of species limits in Neotropical pipits Anthus based on multilocus genetic and vocal data." Ibis.

Thanks to Paul van Els.

## `babette` misc

### Why the name?

 * `ba`: BeAutier
 * `bet`: BEasTier
 * `te`: TracErer

Later on, `mauricer` got added.

### Why the logo?

Initially, the logo was a low-tech remake of Babette, a maid in Beauty and the Beast. 
To prevent problems with Disney, a different logo was picked.

The current logo shows a swan, an animal considered to be graceful.
The swan is drawn by Jose Scholte, who kindly allowed her work to
be used for free, by attribution.

## Error: `libjvm.so: cannot open shared object file: No such file or directory`

For me, [this Stack Overflow post](https://stackoverflow.com/a/25932828) helped me out:

```
sudo mousepad /etc/ld.so.conf.d/java.conf
```

In that file put:

```
/usr/lib/jvm/java-8-oracle/jre/lib/amd64
/usr/lib/jvm/java-8-oracle/jre/lib/amd64/server
```

Save, close, restart R studio, fixed!

Else [this Stack Overflow post may be helpful](https://stackoverflow.com/a/43466434):

```
sudo apt-get install jdk-*
sudo R CMD javareconf
sudo R CMD javareconf -e
export LD_LIBRARY_PATH=$JAVA_LD_LIBRARY_PATH
sudo apt-get install r-cran-rjava
```
